lib/location/method_gnss.c

```diff
@@ -109,6 +109,54 @@ static void method_gnss_pgps_ext_work_fn(struct k_work *item);
 
 static int fixes_remaining;
 
+static inline int check_modem_supports_gnss(void)
+{
+	int err;
+	enum lte_lc_system_mode mode;
+
+	err = lte_lc_system_mode_get(&mode, NULL);
+	if (err) {
+		LOG_ERR("Could not read modem system mode. Error: %d", err);
+		return err;
+	}
+
+	if (mode != LTE_LC_SYSTEM_MODE_GPS &&
+	    mode != LTE_LC_SYSTEM_MODE_LTEM_GPS &&
+	    mode != LTE_LC_SYSTEM_MODE_NBIOT_GPS &&
+	    mode != LTE_LC_SYSTEM_MODE_LTEM_NBIOT_GPS) {
+		LOG_ERR("The currently selected modem system mode does not support GNSS!");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static inline int check_modem_supports_lte(void)
+{
+	int err;
+	enum lte_lc_system_mode mode;
+
+	err = lte_lc_system_mode_get(&mode, NULL);
```

Comment:
Was thinking if lte_lc_lte_mode_get() would be better to do the same? It is returning "none" in plain gnss mode.

Reply:
Good point. Sorry for the delay on the response, I had another project come up and take precedence.

I'll give this a shot and test thoroughly